woocommerce/woocommerce-android
# Move most CardReaderModule's classes to debug package and add its dependencies only in debug builds

This PR moves classes which are dependant on StripeTerminalSDK into `debug` package. It also replaces `implementation` with `debugImplementation` dependency type in the module's `build.gradle`.
All these changes have been made to make sure that we don't increase the size of the release APK before the CardReader feature gets actually released.

**Merge instructions**
1. Make sure https://github.com/woocommerce/woocommerce-android/pull/3766 is merged
2. Remove "Not ready for merge" label
3. Merge this PR

To test:
1. run `./gradlew :WooCommerce:dependencies --configuration wasabiDebugRuntimeClasspath` and notice `com.stripe:stripeterminal:` is present
2. run `./gradlew :WooCommerce:dependencies --configuration vanillaReleaseRuntimeClasspath` and notice `com.stripe:stripeterminal:` is not present
3. Make sure it's still possible to build a release APK (I wasn't able to verify this since my environment isn't ready for building production apks. Let me know if you'd prefer me to test this as well.)


Update release notes:
